arwen/host/execution.go

```diff
@@ -710,6 +726,9 @@ func (host *vmHost) callSCMethod() error {
 	if err != nil {
 		err = host.handleBreakpointIfAny(err)
 	}
+	if err == nil {
```

Comment:
I think L725 - L734 could be refactored like this to be more clear:

	_, err = function()
	if err != nil {
		err = host.handleBreakpointIfAny(err)
	        if err != nil {
		       return err
	        }
	}

	err = host.checkFinalGasAfterExit()
	if err != nil {
		return err
	}

Reply:
I would leave like this. It is the correct way to do it.